cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,297 @@
+				return fmt.Errorf("config[%d].to[%d].namespace: empty value is not allowed", i, j)
+			}
+			if secretContext.Name == "" {
+				return fmt.Errorf("config[%d].to[%d].name: empty value is not allowed", i, j)
+			}
+			if _, ok := kubeConfigs[secretContext.Cluster]; !ok {
+				return fmt.Errorf("config[%d].to[%d]: failed to find cluster context %q in the kubeconfig", i, j, secretContext.Cluster)
+			}
+
+			kc := kubeConfigs[secretContext.Cluster]
+			client, err := coreclientset.NewForConfig(&kc)
+			if err != nil {
+				return err
+			}
+			o.secretsGetters[secretContext.Cluster] = client
+
+			if toMap[secretContext.Cluster] == nil {
+				toMap[secretContext.Cluster] = map[string]string{secretContext.Namespace: secretContext.Name}
+			} else if toMap[secretContext.Cluster][secretContext.Namespace] != secretContext.Name {
+				toMap[secretContext.Cluster][secretContext.Namespace] = secretContext.Name
+			} else {
+				return fmt.Errorf("config[%d].to[%d]: secret %v showed more than once in the config", i, j, secretContext)
+			}
+		}
+	}
+	return nil
+}
+
+type bitWardenContext struct {
+	BWItem     string `json:"bw_item"`
+	Field      string `json:"field,omitempty"`
+	Attachment string `json:"attachment,omitempty"`
+}
+
+type secretContext struct {
+	Cluster   string `json:"cluster"`
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+type secretConfig struct {
+	From map[string]bitWardenContext `json:"from"`
+	To   []secretContext             `json:"to"`
+}
+
+func getSecrets(config []secretConfig, bwClient bitwarden.Client) (map[string][]*coreapi.Secret, error) {
+	secretsMap := map[string][]*coreapi.Secret{}
+	for _, secretConfig := range config {
+		data := make(map[string][]byte)
+		for key, bwContext := range secretConfig.From {
+			var value []byte
+			var err error
+			if bwContext.Field != "" {
+				itemGetter, err := bwClient.GetItem(bwContext.BWItem)
+				if err != nil {
+					return nil, err
+				}
+				field, err := itemGetter.GetField(bwContext.Field)
+				if err != nil {
+					return nil, err
+				}
+				value = []byte(field)
+			} else {
+				itemGetter, err := bwClient.GetItem(bwContext.BWItem)
+				if err != nil {
+					return nil, err
+				}
+				value, err = itemGetter.GetAttachment(bwContext.Attachment)
+				if err != nil {
+					return nil, err
+				}
+			}
+			if err != nil {
+				return nil, err
+			}
+			data[key] = value
+		}
+		for _, secretContext := range secretConfig.To {
+			secret := &coreapi.Secret{
+				Data: data,
+				ObjectMeta: meta.ObjectMeta{
+					Name:      secretContext.Name,
+					Namespace: secretContext.Namespace,
+					Labels:    map[string]string{"ci.openshift.org/auto-managed": "true"},
+				},
+			}
+			secretsMap[secretContext.Cluster] = append(secretsMap[secretContext.Cluster], secret)
+		}
+	}
+	return secretsMap, nil
+}
+
+func applySecrets(secretsGetters map[string]coreclientset.SecretsGetter, secretsMap map[string][]*coreapi.Secret) error {
```

Comment:
You are not applying secrets here, you are Updating them. IMO we should use Replace. Update can fail if someone else touches them but we know we're the only source of truth here.

Reply:
Renamed.
Keep the update call and see if we hit issues (as discussed).